src/stavisky_lab_to_nwb/redis_interfaces/redisrecordingextractor.py

```diff
@@ -0,0 +1,276 @@
+"""Redis recording extractor."""
+import redis
+import numpy as np
+from typing import Union, Optional, Literal
+
+from spikeinterface.core import BaseRecording, BaseRecordingSegment
+from stavisky_lab_to_nwb.redis_interfaces.redisextractormixin import RedisExtractorMixin
+
+
+class RedisStreamRecordingExtractor(BaseRecording, RedisExtractorMixin):
+    """Recording extractor for recording data stored in Redis stream"""
+
+    def __init__(
+        self,
+        port: int,
+        host: str,
+        stream_name: str,
+        data_key: Union[bytes, str],
+        dtype: Union[str, type, np.dtype],
+        channel_count: int,
+        channel_ids: Optional[list] = None,
+        frames_per_entry: int = 1,
+        start_time: Optional[float] = None,
+        sampling_frequency: Optional[float] = None,
+        timestamp_source: Optional[Union[bytes, str]] = None,
+        timestamp_kwargs: dict = {},
+        gain_to_uv: Optional[float] = None,
+        channel_dim: int = 0,
+    ):
+        """Initialize the RedisStreamRecordingExtractor
+
+        Parameters
+        ----------
+        port : int
+            Port number for Redis server
+        host : str
+            Host name for Redis server, e.g. "localhost"
+        stream_name : str
+            Name of stream containing the recording data
+        data_key : bytes or str
+            Key or field within each Redis stream entry that
+            contains the recording data
+        dtype : str, type, or numpy.dtype
+            The dtype of the data. Assumed to be a numeric type
+            recognized by numpy, e.g. int8, float32, etc.
+        channel_count : int
```

Comment:
Does this need to be the total number of channels from the array in redis? Or is it simply the max of the channel_ids? or the length of the channel IDs?

I can totally understand if it's required information to even read the array from redit, but otherwise it's a little odd to have two inter-related inputs for handling channel specifications

Reply:
This is the total number of channels in the array in Redis. I added it in because `channel_ids` is optional (following what I saw in `spikeinterface` generally) and we'd need to know channel count to autogenerate `channel_ids`. But, now that I think about it, we could infer channel count given `frames_per_entry`, since the total number of entries in each Redis array has to be `frames_per_entry * channel_count`. We'd just have to add an extra step to read the first entry, but that's not expensive